engine/Shopware/Models/Document/Document.php

```diff
@@ -55,6 +55,16 @@ class Document extends ModelEntity
      */
     private $name = '';
 
+    /**
+     * A freely choseable internal key, which can be used to identify a document type independently
+     * form it's id or name, because these values may vary depending of the shop context (installed
+     * plugins) and the user customization.
+     *
+     * @var string
+     * @ORM\Column(name="`key`", type="string", nullable=false, unique=true)
+     */
```

Comment:
That default value doesn't make sense in conjunction with the unique constraint.